### DISCUSSION TITLE
What is the process to update Pages

### DISCUSSION THREAD

**Discussion by raramuridesign** [joomlatools/joomlatools-pages#760] [Q&A] [answered]
Note: referencing this https://github.com/joomlatools/joomlatools-pages/discussions/738#discussioncomment-859025

It would be a good idea to understand how we go about updating the code.
Assume we are not using github and rely on the Joomla Updater. How would we go about updating the code when new features are added. Is this currently

1. a manual process
2. or will the Joomla updater pick this up?

Matthew

@johanjanssens

**Comment by johanjanssens** [comment, marked as answer]
Happy to help you. Short answer, updating can be done in 2 ways:

### 1. Using Joomla installer

This is a manual two step process:

1. Uninstall Joomlatools Framework, which also uninstalls Pages
2. Re-install Pages, which also re-installs Joomlatools Framework

_Note:_ There is an open ticket to add auto-updated: https://github.com/joomlatools/joomlatools-pages/issues/211 I considering this more a nice to have then a need to have. Other higher priority tasks I like to work on first. 

### 2. Using composer (recommended)

The recommended way to keep Pages updated is by installing it using composer and updating it using composer.  This is also the only way you can install Pages when running it standalone.

For the more in-depth answer keep reading.

## When to update Pages

Pages is not an end user extension, it doesn't offer any "features". It a technical tool for integrators and developers. The traditional way of thinking about a Joomla extension doesn't apply to Pages. You don't need to update Pages with each release. You should update when it makes sense. 

Lets consider the different reasons when and why you will update Pages:

### 1. Security

> Because there is a security issue in the code and an update is required to resolve this security problem. 

Security issues in Pages will be rare if none existing. Pages is a frontend component, the only thing it does is render data, it doesn't process user input, besides from the forms support it has. As such it's not very prone to security problems. 

### 2. Compatibility

> Because the code is no longer compatible with the hosting environment you use, and/or the underlying technology is deprecated. For example, updates in PHP versions and end of life of older versions might require you to update. 

Compatibility issues will not be an issue for a few years at least, Pages sets the minimal PHP version to 7.3 now and is already forward compat with PHP8. This gives the current code at lifetime of 3-5 years at least.

### 3. Feature

> Because a new version brings a new feature you like to make use off. 

Updates to make use of a new feature will require additional developer work, out of the box the new feature will not do anything until it's implemented. This means you will do this as part of a site development or redevelopment, and you will very likely discuss this with the owner of the site. 

Taking the above into account, updating Pages will be a deliberate action you take. It's not something you will do or need to do on a regular basis. Once Pages is installed and used in a production site there will not be any reason to update it unless for any of these 3 reasons.

**Reply by raramuridesign** [reply]
@johanjanssens Thanks for the clarification on this.
